plugins/ci/pkg/ci/trivy.go

```diff
@@ -20,40 +20,47 @@ import (
 	"github.com/fairwindsops/insights-plugins/plugins/ci/pkg/commands"
 	"github.com/fairwindsops/insights-plugins/plugins/ci/pkg/models"
 	ciutil "github.com/fairwindsops/insights-plugins/plugins/ci/pkg/util"
+	"github.com/hashicorp/go-multierror"
 )
 
-func (ci *CIScan) GetTrivyReport(dockerImages []trivymodels.DockerImage, manifestImages []trivymodels.Image) (*models.ReportInfo, error) {
+func (ci *CIScan) GetTrivyReport(dockerImages []trivymodels.DockerImage, manifestImages []trivymodels.Image) (report *models.ReportInfo, errs error) {
+	allErrs := new(multierror.Error)
 	err := updatePullRef(ci.config.Images.FolderName, dockerImages, manifestImages)
 	if err != nil {
 		return nil, err
 	}
 
 	filenameToImageName, err := downloadMissingImages(ci.config.Images.FolderName, dockerImages, manifestImages, ci.config.Options.RegistryCredentials)
 	if err != nil {
-		return nil, err
+		allErrs = multierror.Append(allErrs, err)
 	}
 
 	allImages, err := mergeImages(ci.config.Images.FolderName, dockerImages, manifestImages, filenameToImageName)
```

Comment:
Should a failed call to `MergeImages` fail the trivy scan completely, or could the scan potentially continue?

Reply:
Seems like the only thing that may fail is walking through folders, I think it is doable to partially process using images that were processed before the error occurs...